rules/legacy_xcodeproj.bzl

```diff
@@ -332,7 +335,7 @@ def _xcodeproj_aspect_impl(target, ctx):
                 asset_srcs += getattr(ctx.rule.files, attr, [])
         srcs = [f for f in srcs if _is_current_project_file(f)]
         non_arc_srcs = [f for f in non_arc_srcs if _is_current_project_file(f)]
-        asset_srcs = [f for f in asset_srcs if _is_current_project_file(f)]
+        asset_srcs = [f for f in asset_srcs if _is_current_package_file(f, target.label.package)]
```

Comment:
@larrytaowang I haven't fully looked at this but remembered there was 1-2 changes we needed. This said, at this point in the program - you have already a looped assets. You might dig into _why the upstream_ [code](https://github.com/bazel-ios/rules_ios/pull/578/files#diff-a15f63f3a76f05dbdfb6a9b817d2d5b1dfd28aa94bafc362eca9bd53f0dd77e4L332) processes it like this: looping `ctx.rule.files` in this way: e.g. why should it loop irrelevant sources and filter them down - better to fix at the root if possible IMO.

> Without VFS, in the xcodegen.json file, the source files of pod A is only included in the generated target A. With VFS, however, the source files of pod A are included in both target A and target B.

Reply:
good question.  I think there is nothing wrong with the upstream code?

In the old way, the file `<generated file Pods/A/A/A.framework/Headers/foo.h>` is referencing an intermediate file, so it is generated. For VFS, the file `<source file Pods/A/A/foo.h>` is referencing a source file in the codebase, so it is a source file.